### PR TITLE
[WFCORE-4446] Migrating AbsolutePathService & RelativePathService to new MSC API

### PR DESCRIPTION
**Pull request by ropalka**
https://issues.jboss.org/browse/WFCORE-4446

### CODE REVIEW

File: controller/src/main/java/org/jboss/as/controller/services/path/AbsolutePathService.java [hunk 1]
```diff
@@ -22,50 +22,47 @@
...
-        AbsolutePathService service = new AbsolutePathService(abstractPath);
-        ServiceBuilder<String> builder = serviceTarget.addService(sname, service);
+    public static ServiceController<?> addService(final ServiceName name, final String abstractPath, final ServiceTarget target) {
+        final ServiceBuilder<?> builder = target.addService(name);
+        final Consumer<String> pathConsumer = builder.provides(name);
+        builder.setInstance(new AbsolutePathService(abstractPath, pathConsumer));
         return builder.install();
     }
 
-    public static String convertPath(String abstractPath) {
```
Comment (jmesnil) [inline controller/src/main/java/org/jboss/as/controller/services/path/AbsolutePathService.java]: Is this public static method considered as API for this class?

File: controller/src/main/java/org/jboss/as/controller/services/path/AbsolutePathService.java [hunk 2]
```diff
@@ -22,50 +22,47 @@
...
+    static String convertPath(final String abstractPath) {
         Assert.checkNotNullParam("abstractPath", abstractPath);
         Assert.checkNotEmptyParam("abstractPath", abstractPath);
         // Use File.getAbsolutePath() to make relative paths absolute
         File f = new File(abstractPath);
         return f.getAbsolutePath();
     }
 
-    public static void addService(final ServiceName name, final ModelNode element, final ServiceTarget serviceTarget) {
```
Comment (jmesnil) [inline controller/src/main/java/org/jboss/as/controller/services/path/AbsolutePathService.java]: same question as above?

File: controller/src/main/java/org/jboss/as/controller/services/path/PathManagerService.java [hunk 3]
```diff
@@ -265,18 +265,12 @@ final void removePathEntry(final String pathName, boolean check) throws Operatio
      * @param pathName the name of the relevant path. Cannot be {@code null}
      * @param path  the value of the path within the model. This is either an absolute path or
      *              the relative portion of the path. Cannot be {@code null}
-     * @param possiblyAbsolute {@code true} if the path may be absolute and a check should be performed before installing
-     *                         a service variant that depends on the service associated with {@code relativeTo}
      * @param relativeTo the name of the path this path is relative to. If {@code null} this is an absolute path
      * @return the service controller for the {@code Service<String>}
      */
     final ServiceController<?> addRelativePathService(final ServiceTarget serviceTarget, final String pathName, final String path,
-                                                      final boolean possiblyAbsolute, final String relativeTo) {
```
Comment (jmesnil) [inline controller/src/main/java/org/jboss/as/controller/services/path/PathManagerService.java]: This seems like a API breaking change.
The removal of the `possiblyAbsolute` changes the behaviour of the method. Without it, it is no longer possible for callers to get an AbsolutePathService from this method.

**Comment by ropalka** [comment]
All fixed @jmesnil

**Pull request merged**